src/hotspot/cpu/ppc/sharedRuntime_ppc.cpp

```diff
@@ -1657,12 +1657,12 @@ static void continuation_enter_cleanup(MacroAssembler* masm) {
   if (CheckJNICalls) {
     // Check if this is a virtual thread continuation
     Label L_skip_vthread_code;
-    __ lwz(R0, in_bytes(ContinuationEntry::flags_offset()), R1_SP);
+    __ ld(R0, in_bytes(ContinuationEntry::flags_offset()), R1_SP);
     __ cmpwi(CCR0, R0, 0);
     __ beq(CCR0, L_skip_vthread_code);
 
     Label L_no_warn;
-    __ lwz(R0, in_bytes(JavaThread::jni_monitor_count_offset()), R16_thread);
+    __ ld(R0, in_bytes(JavaThread::jni_monitor_count_offset()), R16_thread);
     __ cmpwi(CCR0, R0, 0);
```

Comment:
This should be a `cmpdi` since `_jni_monitor_count` is an `intx`.

Reply:
Good catch! Fixed. Thanks